From: Armin Novak <armin.novak@thincast.com>
Date: Tue, 1 Feb 2022 11:49:12 +0100
Subject: Removed cached wlog pointer

(cherry picked from commit 77580466ae31c93af63c55e4cb56b297c01093f3)
(cherry picked from commit 0089ce25b48535e081e9680bf4f2bb9d7e0d0534)
---
 winpr/include/winpr/assert.h | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/winpr/include/winpr/assert.h b/winpr/include/winpr/assert.h
index b8530bf..0cef841 100644
--- a/winpr/include/winpr/assert.h
+++ b/winpr/include/winpr/assert.h
@@ -33,9 +33,7 @@
 	{                                                                                         \
 		if (!(cond))                                                                          \
 		{                                                                                     \
-			static wLog* _log_cached_ptr = NULL;                                              \
-			if (!_log_cached_ptr)                                                             \
-				_log_cached_ptr = WLog_Get("com.freerdp.winpr.assert");                       \
+			wLog* _log_cached_ptr = WLog_Get("com.freerdp.winpr.assert");                     \
 			WLog_Print(_log_cached_ptr, WLOG_FATAL, "%s [%s:%s:%" PRIuz "]", #cond, __FILE__, \
 			           __FUNCTION__, __LINE__);                                               \
 			winpr_log_backtrace_ex(_log_cached_ptr, WLOG_FATAL, 20);                          \
